src/mapgen_singlenode.cpp

```diff
@@ -87,9 +87,10 @@ void MapgenSinglenode::makeChunk(BlockMakeData *data)
 	// Add top and bottom side of water to transforming_liquid queue
 	updateLiquid(&data->transforming_liquid, node_min, node_max);
 
-	// Calculate lighting
-	if (flags & MG_LIGHT)
-		calcLighting(node_min, node_max);
+	// Set lighting
+	u8 light = ((flags & MG_LIGHT) && ndef->get(n_node).sunlight_propagates) ?
```

Comment:
LIGHT_SUN is == 0x0F, so this would only set the day bank.
It needs to be ((LIGHT_SUN << 8) | LIGHT_SUN) (aka 0xFF)